frontend/openchat-agent/src/services/openchatAgent.ts

```diff
@@ -3108,4 +3112,38 @@ export class OpenChatAgent extends EventTarget {
     reportedMessages(userId: string | undefined): Promise<string> {
         return this._userIndexClient.reportedMessages(userId);
     }
+
+    acceptP2PSwap(chatId: ChatIdentifier, threadRootMessageIndex: number | undefined, messageId: bigint): Promise<AcceptP2PSwapResponse> {
```

Comment:
I personally think it is fairly mad (legally) to offer this feature a) at all and b) in groups and channels.